build/config.cmake

```diff
@@ -80,15 +49,27 @@ SET (ESCARGOT_CXXFLAGS_COMMON
     -Wno-implicit-fallthrough
 )
 
-IF (NOT ${ESCARGOT_HOST} STREQUAL "darwin")
-    SET (ESCARGOT_CXXFLAGS_COMMON ${ESCARGOT_CXXFLAGS_COMMON} -frounding-math -fsignaling-nans)
-    SET (ESCARGOT_CXXFLAGS_COMMON ${ESCARGOT_CXXFLAGS_COMMON} -Wno-unused-but-set-variable -Wno-unused-but-set-parameter)
+IF (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+    SET (ESCARGOT_CXXFLAGS_COMMON ${ESCARGOT_CXXFLAGS_COMMON} -frounding-math -fsignaling-nans -Wno-unused-but-set-variable -Wno-unused-but-set-parameter -Wno-unused-parameter)
+ELSEIF (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+    SET (ESCARGOT_CXXFLAGS_COMMON ${ESCARGOT_CXXFLAGS_COMMON} -fno-fast-math -fno-unsafe-math-optimizations -fdenormal-fp-math=ieee -Wno-parentheses-equality -Wno-unused-parameter -Wno-dynamic-class-memaccess -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type)
+ELSE()
+    MESSAGE (FATAL_ERROR ${CMAKE_CXX_COMPILER_ID} " is Unsupported Compiler")
 ENDIF()
 
 
 # ESCARGOT COMMON LDFLAGS
 SET (ESCARGOT_LDFLAGS_COMMON -fvisibility=hidden)
 
+IF (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+    SET (ESCARGOT_LDFLAGS_COMMON ${ESCARGOT_LDFLAGS_COMMON})
+ELSEIF (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+    SET (ESCARGOT_LDFLAGS_COMMON ${ESCARGOT_LDFLAGS_COMMON})
```

Comment:
There are no other flags depend on the compiler. Is it intended or are you going to update it later? Other is good to me.